src/controllers/hid/hidcontroller.h

```diff
@@ -60,45 +57,79 @@ class HidControllerJSProxy : public ControllerJSProxy {
               m_pHidController(m_pController) {
     }
 
-    Q_INVOKABLE void send(const QList<int>& data, unsigned int length = 0) override {
-        m_pHidController->send(data, length);
+    /// @brief Sends HID OutputReport with hard coded ReportID 0 to HID device
+    ///        This function only works with HID devices, which don't use ReportIDs
+    /// @param dataList Data to send as list of bytes
+    /// @param length Unused optional argument
+    Q_INVOKABLE void send(const QList<int>& dataList, unsigned int length = 0) override {
```

Comment:
How does this comment relate: 
https://github.com/mixxxdj/mixxx/blob/f2d670edeb483efb066bf7f00255d01df2f0017c/res/controllers/common-hid-packet-parser.js#L467

Introduced here: 
https://github.com/mixxxdj/mixxx/pull/1953

Reply:
These are different tings:
There are two categories of HID devices:

1. Devices that use Report-IDs in no report (than the device can have only one report of each type Input/Output/Feature).
   hidapi expect the Report-ID byte for Output and Feature-Reports to be 0 in this case, but don't prepend this 0 to Input-Reports itself.

2. Devices that use a Report-ID in any report (allowed range of report IDs is 1...255).

Therefore you can use the sendOutputReport/sendFeatureReport functions and specify the Report-ID argument as 0, if you want send data to an HID device, that don't uses Report-IDs.

Reply:
Ah OK, that makes sense.